utils/unittest/network.go

```diff
@@ -1,25 +1,99 @@
 package unittest
 
 import (
+	"fmt"
+
 	"github.com/stretchr/testify/mock"
 
-	module "github.com/onflow/flow-go/module/mock"
-	netint "github.com/onflow/flow-go/network"
+	"github.com/onflow/flow-go/model/flow"
+	mockmodule "github.com/onflow/flow-go/module/mock"
+	"github.com/onflow/flow-go/network"
 	"github.com/onflow/flow-go/network/mocknetwork"
 )
 
-// RegisterNetwork returns a mocked network and conduit
-func RegisterNetwork() (*module.Network, *mocknetwork.Conduit) {
-	con := &mocknetwork.Conduit{}
+type EngineProcessFunc func(network.Channel, flow.Identifier, interface{}) error
+type NetworkPublishFunc func(network.Channel, interface{}, ...flow.Identifier) error
+
+// Conduit represents a mock conduit.
+type Conduit struct {
+	mocknetwork.Conduit
+	net     *Network
+	channel network.Channel
+}
+
+// Publish sends a message on this mock network, invoking any callback that has
+// been specified. This will panic if no callback is found.
+func (c *Conduit) Publish(event interface{}, targetIDs ...flow.Identifier) error {
+	if c.net.publishFunc != nil {
+		return c.net.publishFunc(c.channel, event, targetIDs...)
+	}
+	panic("Publish called but no callback function was found.")
+}
+
+// Network represents a mock network. The implementation is not concurrency-safe.
```

Comment:
Nit: it would be a cheap fix to make it so, simply locking additions to conduits and engines in `Register` and read-locking in `Send`

Reply:
I'll definitely add it in if there is a usecase for it in the future :)